docs/operations/versioning.md

```diff
@@ -0,0 +1,305 @@
+    version 1.17.0 will deprecated with the expiration date for `supported` versions. 
+
+* * *
+***A new minor release version has been released and should be added to the CloudProfile.***
+
+**Example**: Kubernetes releases version 1.19.0.
+
+ - Deprecate the current `supported` version, optionally with an expiration date. Afterwards add the new version as a `preview` version.
+ 
+ - `Version Management` enabled: Add version 1.19.0 without a classification. 1.19.0 will be classified as `supported` and version 1.18.3 will be expired with the configured expiration date.
+ 
+ - With `Version Maintenance` enabled: Add version 1.19.0 without a classification. `Version Maintenance` makes sure that versions now having an `unmaintained` minor, will be accordingly deprecated. 1.19.0 will be classified as `supported` and version 1.16.0 will be deprecated having no expiration date.
+ 
+* * *
+***A `preview` version should be promoted to `supported`.***
+
+**Example**: Kubernetes version 1.18.4 should be promoted to `supported`.
+
+ - Deprecate the currently `supported` version 1.18.3, optionally with a expiration date. Afterwards change the classification of the version 1.18.4 from `preview` to `supported`.
+ 
+ - `Version Management` enabled: Deprecate the current `supported` version 1.18.3. An expiration date is being set automatically. Afterwards change the classification of the version 1.18.4 from `preview` to `supported`
+ 
+ - With `Version Maintenance` enabled: Same as for `Version Management`, but make sure to deprecate the `supported` version of the particular minor version (in this example: 1.18.3).
+ 
+* * *
+***The `supported` version should be deprecated.***
+
+**Example**: Version 1.18.3 should be deprecated.
+
+ - Deprecate version 1.18.3 by setting the classification `deprecated`, optionally with a expiration date.
+ 
+ - `Version Management` enabled: By just adding a new version e.g 1.18.4 without a classification, Version 1.18.3 will be deprecated and 1.18.4 becomes the new `supported` version.  
+   However, is there no higher non-preview version available, version 1.18.3 will be reset to `supported` without an expiration date. Having one `supported` version is enforced by the `Version Management`.
+ 
+ - With `Version Maintenance` enabled:  Similar as for `Version Management`. Having one `supported` version per `maintained` minor version is enforced. 
+ 
+* * *
+**The current `supported` version has problems and should be replaced with the previous `supported` version.**
+
+**Example**: Version 1.18.3 has problems.
+
+a) Version is not in use by any Shoot.
+ - delete the version from the CloudProfile. 
+ - Manually change the classification of the previously `supported` version.
+ - `Version Management` and `Version Maintenance`: nothing else has to be done. 
+
+b) Shoots are already running the erroneous version
+ -  Make sure there is a higher non-preview version available in the CloudProfile.
```

Comment:
```suggestion
 -  Make sure there is a higher non-preview version available in the `CloudProfile`.
```

Reply:
What if this is not the case? For example, when 1.18.4 is not yet released?

Reply:
see [below describing that usecase.](https://github.com/gardener/gardener/blob/aaca736472c700f9cd9f8b4dbec06b9d2f4f90c0/docs/operations/versioning.md#L291)